ginivision/src/androidTest/java/net/gini/android/vision/test/JpegByteArraySubject.java

```diff
@@ -0,0 +1,120 @@
+package net.gini.android.vision.test;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.common.truth.FailureStrategy;
+import com.google.common.truth.Subject;
+import com.google.common.truth.SubjectFactory;
+
+import net.gini.android.vision.test.exif.ExifReader;
+import net.gini.android.vision.test.exif.ExifReaderException;
+
+class JpegByteArraySubject extends Subject<JpegByteArraySubject, byte[]> {
+
+    // Not true that <"subject"> "verb" <"expected">. "failure message".
+    private static final String RAW_MESSAGE_TEMPLATE = "Not true that <%s> %s <%s>. %s.";
+
+    static SubjectFactory<JpegByteArraySubject, byte[]> jpegByteArray() {
+        return new SubjectFactory<JpegByteArraySubject, byte[]>() {
+
+            @Override
+            public JpegByteArraySubject getSubject(final FailureStrategy fs, final byte[] that) {
+                return new JpegByteArraySubject(fs, that);
+            }
+        };
+    }
+
+    JpegByteArraySubject(@NonNull final FailureStrategy failureStrategy,
+            @Nullable final byte[] subject) {
+        super(failureStrategy, subject);
+    }
+
+    void hasUUIDinUserComment(@Nullable  String uuid) {
+        isNotNull();
+        String verb = "has in User Comment UUID";
+
+        if (uuid == null) {
+            fail(verb, (Object) null);
+            return;
+        }
+
+        final String userComment = readExifUserComment(verb, uuid, getSubject());
+        final String uuidInUserComment = getValueForKeyfromUserComment("UUID", userComment);
+
+        if (uuidInUserComment == null) {
+            failWithRawMessage(RAW_MESSAGE_TEMPLATE, getSubject(), verb, uuid,
+                    "It had no UUID in User Comment");
+            return;
+        }
+
+        if (!uuid.equals(uuidInUserComment)) {
+            failWithBadResults(verb, uuid, "was", uuidInUserComment);
+        }
+    }
+
+    @NonNull
+    private String readExifUserComment(@NonNull final String verb, @NonNull final String expected, @NonNull final byte[] jpeg) {
+        try {
+            ExifReader reader = new ExifReader(jpeg);
+            return reader.getUserComment();
+        } catch (ExifReaderException e) {
+            failWithRawMessage(RAW_MESSAGE_TEMPLATE, getSubject(), verb, expected,
+                    "An error occurred: " + e.getMessage());
+        }
+        return "";
+    }
+
+    @NonNull
+    private String readExifUserComment(@NonNull final byte[] jpeg) {
+        try {
+            ExifReader reader = new ExifReader(jpeg);
+            return reader.getUserComment();
+        } catch (ExifReaderException e) {
+            failWithRawMessage("Could not read User Comment from <%s> due to error: %s", jpeg, e.getMessage());
```

Comment:
This class is an assertion extension for Truth. One has to make assertions here. Is used by the `PhotoSubject` and `DocumentSubject` assertion extension classes as they both contain a jpeg byte array. 

These improve readability of test cases, for ex.: `assertAbout(photo()).that(photo).hasUUIDinUserComment(photo.getUUID());`.